publish.py

```diff
@@ -52,189 +51,120 @@ def slugify(name):
     return re.sub(r'\s', '-', out)
 
 
-def create_item(item):
-    """
-    Takes a hacker news json object and creates a hugo markdown page from it
-    :param item: dict of article data
-    """
-    slug = slugify(item.get('title'))
-    file_name = './content/en/post/{}.md'.format(slug)
-    with open(file=file_name, mode='w', encoding='utf-8') as f:
-        item['date'] = datetime.fromtimestamp(item.get('time')).isoformat()
-        item['linkurl'] = item.get('url')
-        item['slug'] = slug
-        item['tags'] = []
-        item['categories'] = []
-        if item.get('time', None):
-            del item['time']
-        if item.get('url', None):
-            del item['url']
-        if item.get('type', None):
-            item['categories'] = ["{}".format(item.get('type'))]
-            del item['type']
-        f.write(TEMPLATE.format(front_matter=yaml.dump(item).strip(), content=""))
-
-
-def create_comment(comment):
-    """
-    Takes a hacker news json object comment and creates a hugo data yaml file for it
-    :param comment: dict of comment data
-    """
-    if comment:
-        item_id = comment.get('id')
-        file_name = f'./data/post/{item_id}.yaml'
-        with open(file=file_name, mode='w', encoding='utf-8') as f:
-            f.write(yaml.dump(comment))
+async def write_file(data):
+    if data:
+        file_name, data_out = '', ''
+        data_id = data.get('id')
+        data_type = data.get('type')
+        if data_type in ('story', 'job'):
+            slug = slugify(data.get('title'))
+            data['date'] = datetime.fromtimestamp(data.get('time')).isoformat()
+            data['linkurl'] = data.get('url')
+            data['slug'] = slug
+            data['tags'] = []
+            data['categories'] = []
+            if data.get('time', None):
+                del data['time']
+            if data.get('url', None):
+                del data['url']
+            if data.get('type', None):
+                data['categories'] = ["{}".format(data.get('type'))]
+                del data['type']
```

Comment:
Can we condense these deletes into something more elegant?
e.g this below or a list comprehension or something
```
for x in ('time', 'url', 'type'):
    del data[x]
```